dm/dm/master/server.go

```diff
@@ -1642,6 +1652,25 @@ func (s *Server) waitOperationOk(
 				} else {
 					return false, "", queryResp, terror.ErrMasterOperRespNotSuccess.Generate("relay is disabled for this source")
 				}
+			case *pb.OperateRelayRequest:
+				// including the situation that source status is nil and relay status is nil
+				relayStatus := queryResp.GetSourceStatus().GetRelayStatus()
+				if expect == pb.Stage_Stopped && relayStatus == nil {
```

Comment:
seems  we loss the `relayStatus==nil && expect != pb. Stage_Stopped` branch ? 

although ths may not hanppen with the current logic,i think we should still take this branch into thought in case we go into it later when refactor the relay logic

how about change the if-else to

```go
if relayStatus == ni{
  if except == xxx{
  }

}else{
  if except == xxx{
  }
}
```